[8.2] [DOCS] Field browser and runtime fields on Alerts table (55a23a85) (#1908)
elastic/security-docs: docs/detections/alerts-runtime-fields.asciidoc

+
[role="screenshot"]
image::images/fields-browser.png[Fields browser]

. Click *Create field*. The *Create field* flyout opens.
+
[role="screenshot"]
image::images/create-field-flyout.png[Create field flyout]

. Enter a *Name* for the new field.

. Select a *Type* for the field's data type.

. Turn on the *Set value* toggle and enter a {ref}/modules-scripting-painless.html[Painless script] to define the field's value. The script must match the selected *Type*. For more on adding fields and Painless scripting examples, refer to {kibana-ref}/managing-data-views.html#runtime-fields[Explore your data with runtime fields].

. Use the *Preview* to help you build the script so that it returns the expected field value.

. Configure other field settings as needed. 
+
NOTE: Some runtime field settings, such as custom labels and display formats, display in other areas of {kib} but may not display in the {security-app}. 

. Click *Save*.

[[manage-runtime-fields]]
=== Manage runtime fields

From the *Alerts* table, you can edit the settings for runtime fields, or delete runtime fields.

> . Open the *Fields* browser (*Detect* -> *Alerts* -> *Fields*), then use the *Categories* drop-down filter to display all runtime fields.

Reviewer: The Categories dropdown will not display all runtime fields. Rather, they can click the "Runtime" column to sort by runtime fields. 
<img width="927" alt="Screen Shot 2022-04-26 at 3 07 26 PM" src="https://user-images.githubusercontent.com/6935300/165392667-46bea3df-03db-4223-8088-2994fc53d08b.png">

They can search Category for the "category" of their field, which would be the first word
<img width="930" alt="Screen Shot 2022-04-26 at 3 08 16 PM" src="https://user-images.githubusercontent.com/6935300/165392688-528edb06-ef2e-4524-913c-09c99e86c3f9.png">

Author: Thanks @stephmilovic! I got mixed up when I was testing this because I named all my fields `runtime.x`, which meant "runtime" appeared as a filter option. I'll adjust instructions to use the **Runtime** column header instead.

[[alerts-runtime-fields]]
== Add runtime fields to detection alerts

> Runtime fields are fields that you can add to documents after you've ingested your data, and they are evaluated at query time. For example, you could combine two fields and treat them as one, or perform calculations on existing data and use the result as a separate field.

Reviewer 2: ```suggestion
Runtime fields are fields that you can add to documents after you've ingested your data. For example, you could combine two fields and treat them as one, or perform calculations on existing data and use the result as a separate field. Runtime fields are fields evaluated when a query is run. 
```

> You can add a runtime field to your detection alerts from the *Alerts* table. The new field becomes available on all alerts within {elastic-sec}.

Reviewer 2: ```suggestion
You can add a runtime field to your detection alerts from the *Alerts* table. The new field applies to all alerts in {elastic-sec}.
```

Author: Slight edit, so it doesn't sound like the new field is only applicable _inside_ the Security app:
```suggestion
You can add a runtime field to your detection alerts from the *Alerts* table. The new field applies to all {elastic-sec} alerts.
```

To add a runtime field to detection alerts:

. Go to *Detect* -> *Alerts*, then click the *Fields* toolbar button at the upper-left of the *Alerts* table. The *Fields* browser opens.
+
[role="screenshot"]
image::images/fields-browser.png[Fields browser]

. Click *Create field*. The *Create field* flyout opens.
+
[role="screenshot"]
image::images/create-field-flyout.png[Create field flyout]

. Enter a *Name* for the new field.

. Select a *Type* for the field's data type.

. Turn on the *Set value* toggle and enter a {ref}/modules-scripting-painless.html[Painless script] to define the field's value. The script must match the selected *Type*. For more on adding fields and Painless scripting examples, refer to {kibana-ref}/managing-data-views.html#runtime-fields[Explore your data with runtime fields].

. Use the *Preview* to help you build the script so that it returns the expected field value.

. Configure other field settings as needed. 
+
NOTE: Some runtime field settings, such as custom labels and display formats, display in other areas of {kib} but may not display in the {security-app}. 

. Click *Save*.

[[manage-runtime-fields]]
=== Manage runtime fields

> From the *Alerts* table, you can edit the settings for runtime fields, or delete runtime fields.

Reviewer 2: ```suggestion
From the *Alerts* table, you can manage runtime fields. 
```
